### PR TITLE
Misc i18n fixes

### PR DESCRIPTION
**Pull request by rhymes**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This all started while I was reading the code for #14620 and after installing `main` locally I noticed the following post install message:

```
# Install default configuration:
cp $(bundle exec i18n-tasks gem-path)/templates/config/i18n-tasks.yml config/
# Add an RSpec for missing and unused keys:
cp $(bundle exec i18n-tasks gem-path)/templates/rspec/i18n_spec.rb spec/
```

thus I added the missing file (the spec) and went down a rabbit hole of micro fixes:

- Added the spec file itself
- Ignored the `storybook-static` folder which was not ignored by `i18n-tasks`
- Ran `normalize` on the files (I tried to add it to `lint-staged` but it's not compatible, it doesn't support the absolute path of files, it considers it the name of an additional locale and bails)
- Fixed inconsistent interpolations
- Added some missing french Devise translations and updated the Devise english translations
- Ignored some missking keys and disabled the unused keys test until we have a strategy for i18n

Having a test that breaks should help us limit the bugs we add during the effort to internationalize and localize.

Let me know if you disagree on any of the changes.

ps. `i18n-tasks normalize` sorts keys, that's why we have so many diffs in the files

## Related Tickets & Documents

#14620 

## QA Instructions, Screenshots, Recordings

Click around, nothing should be broken

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

### CODE REVIEW

File: config/locales/devise_invitable.en.yml [hunk 1]
```diff
@@ -1,28 +1,29 @@
+---
```
Comment (rhymes) [inline config/locales/devise_invitable.en.yml:1]: Same argument as above, the only point in having custom english files is to override the text of the english. Here we only customize `devise.edit.submit_button` over the original, so we might as well only have that key and not the whole file - see https://raw.githubusercontent.com/scambra/devise_invitable/v2.0.5/config/locales/en.yml

File: config/locales/devise_invitable.fr.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
```
Comment (rhymes) [inline config/locales/devise_invitable.fr.yml:1]: this we need because it's the french copy

File: config/locales/doorkeeper.en.yml [hunk 3]
```diff
@@ -1,152 +1,140 @@
...
+        title: Doorkeeper
       application:
-        title: 'OAuth authorization required'
+        title: OAuth authorization required
+    pre_authorization:
+      status: Pre-authorization
+    scopes:
+      public: Access public profile data and published articles
+      read_articles: Access all articles including drafts
+      write_articles: Access to creating and updating articles
```
Comment (rhymes) [inline config/locales/doorkeeper.en.yml:140]: here we only add these three custom keys, so we might as well keep them separate and use the default locale file inherited from the gem

File: config/locales/en.yml [hunk 4]
```diff
@@ -1,72 +1,43 @@
...
-#
-# Instead, surround them with single quotes.
-#
-# en:
-#   'true': 'foo'
-#
-# To learn more, please read the Rails Internationalization guide
-# available at https://guides.rubyonrails.org/i18n.html.
-
+---
```
Comment (rhymes) [inline config/locales/en.yml:1]: only reordering

File: config/locales/fr.yml [hunk 5]
```diff
@@ -35,6 +35,7 @@ fr:
     year: Anée
   date:
     abbr_month_names:
+    -
```
Comment (rhymes) [inline config/locales/fr.yml:38]: month 0 doesn't exist, the key is needed, see https://github.com/rails/rails/blob/d59e729fbf48858b672814194a180d6ddf795f3d/activesupport/lib/active_support/locale/en.yml#L16

**Review by citizen428 — APPROVED** [review]

**Review by benhalpern — APPROVED** [review]
Looks good to me pending failing test, and if I'm understanding correctly it is essentially working as expected in that regard?
Screen Shot 2021-09-30 at 2 27 01 PM

**Review by Ridhwana — APPROVED** [review]

**Pull request merged**